packages/blueprints-integration/src/package.ts

```diff
@@ -0,0 +1,273 @@
+			checksum?: string
+			checkSumType?: 'sha' | 'md5' | 'whatever'
+		}
+		sources: {
+			containerId: string
+			accessors: {
+				[accessorId: string]:
+					| AccessorOnPackage.LocalFolder
+					| AccessorOnPackage.FileShare
+					| AccessorOnPackage.HTTP
+			}
+		}[]
+
+		sideEffect: {
+			previewContainerId?: string
+			thumbnailContainerId?: string
+		}
+	}
+	export interface ExpectedPackageQuantelClip extends Base {
+		type: PackageType.QUANTEL_CLIP
+		content: {
+			guid?: string
+			title?: string
+		}
+		version: {
+			// @todo: something here?
+		}
+		sources: {
+			containerId: string
+			accessors: { [accessorId: string]: AccessorOnPackage.Quantel }
+		}[]
+
+		sideEffect: {
+			previewContainerId?: string
+			thumbnailContainerId?: string
+		}
+	}
+}
+
+/** A PackageContainer defines a place that contains Packages, that can be read or written to.
+ * For example:
+ *   A PackageContainer could be a folder on a computer that contains media files.
+ *   That folder could be accessed locally (Accessor.LocalFolder)
+ *   and if the folder is shared, by a Accessor.FileShare over the network
+ */
+export interface PackageContainer {
+	/** Short name, for displaying to user */
+	label: string
+
+	/** A list of ways to access the PackageContainer. Note: The accessors are different ways to access THE SAME PackageContainer. */
+	accessors: { [accessorId: string]: Accessor.Any }
+}
+
+/** Defines different ways of accessing a PackageContainer.
+ * For example, a local folder on a computer might be accessed through a LocalFolder and a FileShare
+ */
+// eslint-disable-next-line @typescript-eslint/no-namespace
+export namespace Accessor {
+	export type Any = LocalFolder | FileShare | HTTP | Quantel
+
+	export enum AccessType {
+		LOCAL_FOLDER = 'local_folder',
+		FILE_SHARE = 'file_share',
+		HTTP = 'http',
+
+		QUANTEL = 'quantel',
+
+		CORE_PACKAGE_INFO = 'core_package_info',
+	}
+
+	export interface Base {
+		type: AccessType
+		label: string
+		allowRead: boolean
+		allowWrite: boolean
+	}
+	/** Defines acess to a local folder */
+	export interface LocalFolder extends Base {
+		type: AccessType.LOCAL_FOLDER
+
+		/** Name/id of the resource, this could for example be the computer name. */
+		resourceId?: string // todo: rename?
+
+		/** Path to the folder
+		 * @example 'C:\media\'
+		 */
+		folderPath: string
+	}
+	export interface FileShare extends Base {
+		type: AccessType.FILE_SHARE
+
+		/** Name/Id of the network the share exists on. Used to differ between different local networks. */
+		networkId?: string
+
+		/** Path to a folder on a network-share
+		 * @example '\\192.168.0.1\shared\'
+		 */
+		folderPath: string
+
+		userName?: string
+		password?: string
+	}
+	export interface HTTP extends Base {
+		type: AccessType.HTTP
+
+		/** Base url (url to the host), for example http://myhost.com/fileShare/ */
+		baseUrl: string
+
+		/** Type of request. Defaults to 'get' */
+		method?: 'get' | 'post' | string
```

Comment:
I'm somewhat unsure about this. What does it mean if the `method` is `post`? Can you "get" the contents of the file using a POST request then? The meaning of this is opaque at this point, and given that the current policy on interfaces is "as minimal as possible", I would suggest removing `method`, `headers` & `requestBody`. These are not well defined & can't be supported for the purpose of Hover-scrub and thumbnails (which are currently the only users of HTTP Accessors).

Reply:
Fixed in https://github.com/nrkno/tv-automation-server-core/commit/115ede6d3e9a947eea788c7290b5757b9cb38e26